oct/cli/package/aws.py

```diff
@@ -30,25 +31,51 @@
     is_flag=True,
     help='Mark an AMI created previously as ready.',
 )
+@option(
+    '--tag',
+    '-t',
+    "tags",
+    multiple=True,
+    help='Tag an AMI with the provided key-value pair.',
+)
 @ansible_output_options
 @package_options
 @pass_context
-def ami(context, update_current_stage, mark_ready):
+def ami(context, update_current_stage, mark_ready, tags):
     """
     Package a running AWS EC2 virtual machine.
 
     :param context: Click context
     :param update_current_stage: whether or not to update current stage
     :param mark_ready: whether or not to mark a previous AMI from this instance ready
+    :param tag: tag an AMI with the provided key-value pair
     """
     configuration = context.obj
+
+    ami_tags = {}
+    for tag in tags:
+        if "=" not in tag:
+            print("Invalid tag provided: {} - must be in the form of KEY=VALUE".format(tag), file=sys.stderr)
```

Comment:
So I switched this to `raise UsageError(...)` but the testing framework is not catching those exceptions and thinks the return code is 0. Long story short, I sticked with pure exit() that works.